lib/openstax/aws/deployment_base.rb

```diff
@@ -176,6 +176,16 @@ def stacks
       self.class.stack_ids.map{|id| self.send("#{id}_stack")}
     end
 
+    def status
+      stacks.map do |stack|
+        {
+          name: stack.name,
+          most_recent_status: stack.status,
+          reasons: stack.latest_failed_events
```

Comment:
To future proof this method so that it can later return status on non-stack things (e.g. shared secrets or an overall status string we make up for the deployment), can we scope this to return this information under a `stacks:` key?

```ruby
def status
  {
    stacks: stacks.map do |stack|
      {
        name: stack.name,
        status: stack.status,
        failed_events_since_last_user_event: stack. failed_events_since_last_user_event
      }
    end
  }
```

(see below on rename of that last method call)

Reply:
You also may want to memoize this so subsequent calls don't have to requery everything:

```ruby
def status(reload: false)
  @status = nil if reload
  @status ||= {
    stacks: ...
  }
end
```

Reply:
See my comments on the unified-deployment PR.  Recommend that `status` return a `Status` object, in which case this becomes

```ruby
def status(reload: false)
  @status = nil if reload
  @status ||= Status.new
end
```